Content.Server/_NF/GameRule/NfAdventureRuleSystem.cs

```diff
@@ -108,233 +122,234 @@ private void OnPlayerSpawningEvent(PlayerSpawnCompleteEvent ev)
 
-        if (_map.TryLoad(mapId, cove, out var depotUid6s, new MapLoadOptions
-            {
-                Offset = _random.NextVector2(10000f, 15000f)
-            }))
+        var dungenTypes = _prototypeManager.EnumeratePrototypes<DungeonConfigPrototype>();
+
+        foreach (var dunGen in dungenTypes)
         {
-            if (_prototypeManager.TryIndex<GameMapPrototype>("Cove", out var stationProto))
+            var seed = _random.Next();
+            var offset = _random.NextVector2(3000f, 8500f) * _distanceOffset;
+            if (!_map.TryLoad(_mapId, "/Maps/_NF/Dungeon/spaceplatform.yml", out var grids,
+                    new MapLoadOptions
+                    {
+                        Offset = offset
+                    }))
             {
-                _station.InitializeNewStation(stationProto.Stations["Cove"], depotUid6s);
+                continue;
             }
 
-            var meta = EnsureComp<MetaDataComponent>(depotUid6s[0]);
-            _meta.SetEntityName(depotUid6s[0], "Pirate's Cove", meta);
-            _shuttle.SetIFFColor(depotUid6s[0], lpbravoColor);
-            _shuttle.AddIFFFlag(depotUid6s[0], IFFFlags.HideLabel);
+            var mapGrid = EnsureComp<MapGridComponent>(grids[0]);
+            _shuttle.AddIFFFlag(grids[0], IFFFlags.HideLabel);
+            _console.WriteLine(null, $"dungeon spawned at {offset}");
+            offset = new Vector2i(0, 0);
+
+            //pls fit the grid I beg, this is so hacky
+            //its better now but i think i need to do a normalization pass on the dungeon configs
+            //because they are all offset. confirmed good size grid, just need to fix all the offsets.
+            _dunGen.GenerateDungeon(dunGen, grids[0], mapGrid, (Vector2i) offset, seed);
         }
+    }
 
-        if (_map.TryLoad(mapId, lodge, out var lodgeUids, new MapLoadOptions
-            {
-                Offset = _random.NextVector2(1650f, 3400f)
-            }))
+    private void GenerateDepots(IEnumerable<PointOfInterestPrototype> depotPrototypes)
+    {
+        //For depots, we want them to fill a circular type dystance formula to try to keep them as far apart as possible
+        //Therefore, we will be taking our range properties and treating them as magnitudes of a direction vector divided
+        //by the number of depots set in our corresponding cvar
+        var protoList = depotPrototypes.ToList();
+        var depotCount = _configurationManager.GetCVar(NF14CVars.CargoDepots);
+        var rotation = 2 * Math.PI / depotCount;
+        var rotationOffset = _random.NextAngle() / depotCount;
+
+        for (int i = 0; i < depotCount && protoList.Count > 0; i++)
         {
-            if (_prototypeManager.TryIndex<GameMapPrototype>("Lodge", out var stationProto))
+            var proto = _random.Pick(protoList);
+            Vector2i offset = new Vector2i(_random.Next(proto.RangeMin, proto.RangeMax), 0);
+            offset.Rotate(rotationOffset);
```

Comment:
Rotate returns a vector, offset isn't actually modified, this suggestion works as expected.
```suggestion
            offset = offset.Rotate(rotationOffset);
```